File: web/src/main/java/no.nav.sbl/rest/EventRessurs.java
package no.nav.sbl.rest;

import no.nav.sbl.rest.domain.RSEvents;
import no.nav.sbl.service.EventService;
import org.springframework.stereotype.Controller;

import javax.inject.Inject;
import javax.ws.rs.*;

import static javax.ws.rs.core.MediaType.APPLICATION_JSON;

@Controller
@Path("/events")
@Consumes(APPLICATION_JSON)
@Produces(APPLICATION_JSON)
public class EventRessurs {

    @Inject
    private EventService eventService;

    @GET
    @Path("{eventId}")
    public RSEvents hentNyeEvents(@PathParam("eventId") String eventId) {
        return new RSEvents().withEvents(eventService.hentEventerEtterId(Long.parseLong(eventId)));
    }
}

File: src/main/java/no/nav/sbl/rest/EventRessurs.kt
package no.nav.sbl.rest

import no.nav.sbl.rest.domain.RSEvents
import no.nav.sbl.service.EventService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/api/events", "/modiacontextholder/api/events"])
class EventRessurs {
    @Autowired
    lateinit var eventService: EventService

    @GetMapping("{eventId}")
    fun hentNyeEvents(
        @PathVariable("eventId") eventId: String,
    ): RSEvents =
        RSEvents(
            events = eventService.hentEventerEtterId(eventId.toLong()),
        )
}
